core/src/main/scala/org/apache/spark/util/ManualClock.scala

```diff
@@ -57,9 +59,19 @@ private[spark] class ManualClock(private var time: Long) extends Clock {
    * @return current time reported by the clock when waiting finishes
    */
   def waitTillTime(targetTime: Long): Long = synchronized {
-    while (time < targetTime) {
-      wait(10)
+    _isWaiting = true
+    try {
+      while (time < targetTime) {
+        wait(10)
+      }
+      getTimeMillis()
+    } finally {
+      _isWaiting = false
     }
-    getTimeMillis()
   }
+
+  /**
+   * Returns whether there is any thread being blocked in `waitTillTime`.
+   */
+  def isWaiting: Boolean = synchronized { _isWaiting }
```

Comment:
Can this ever return `true`, given that its `synchronized` with the only method that mutates `_isWaiting` (and that method sets it to `false` in a finally)?